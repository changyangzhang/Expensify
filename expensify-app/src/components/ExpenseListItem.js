import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = ({id, description, amount, createdAt }) => (
        <Link to={`/edit/${id}`} className="list-item">
            <div>
                <h3 className="list-item__title">{description}</h3>
                <span className="list-item-__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
            </div>
            <div>
                <h3>{numeral(amount/100).format('0,0.00') + ' kr'}</h3>
            </div>
        </Link>
);

export default ExpenseListItem;