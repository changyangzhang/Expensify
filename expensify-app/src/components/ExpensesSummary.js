import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total'
import {Link} from "react-router-dom";

export const ExpensesSummary = ({expenseCount, expensesTotal,uid}) =>{
    const expenseWord = expenseCount === 1 ? 'expense': 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal/100).format('0,0.00') + ' kr';
    return (
        <div className="page-header">
            <div className="content-container">
                {uid === '6LjG2B7bdPazarNgZs5zan08XZg1' && <h1 className="page-header__title"> Welcome Sweet Boy! </h1>}
                <h1 className="page-header__title"> Viewing <span>{expenseCount}</span> {expenseWord} totalling <span> {formattedExpensesTotal} </span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
       </div>
    );
};

const mapStateToProps = (state) =>{
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    const uid = state.auth.uid;

    return{
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        uid : uid
    }
};

export default connect(mapStateToProps)(ExpensesSummary);