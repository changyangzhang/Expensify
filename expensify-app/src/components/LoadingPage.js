import React from 'react';

const LoadingPage= () =>(
    <div className="loader">
        <img className="loader__image" src="images/load.gif" />
    </div>
);

export default LoadingPage;