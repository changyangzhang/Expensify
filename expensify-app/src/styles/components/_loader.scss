.loader{
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vm;
}
.loader__image{
  height: 15rem;
  width: 15rem;
}