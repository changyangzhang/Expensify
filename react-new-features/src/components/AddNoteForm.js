import React, { useState, useContext } from "react";
import NotesContext from '../context/notes-contect';

const AddNoteForm = () =>{
    const { dispatch } = useContext(NotesContext);
    const [title, setTitle] = useState('');
    const [body, setBody] = useState('');

    const addNote = (e) => {
        e.preventDefault();
        // setNotes([
        //     ...notes,
        //     { title, body }
        // ]);
        dispatch({
            type: 'ADD_NOTE',
            title,
            body
        });
        setTitle('');
        setBody('')
    };

    return (
        <>
        <p>Add note</p>
        <form onSubmit={addNote}>
            <input value={title} onChange={(e) => setTitle(e.target.value)}/>
            <textarea value={body} onChange={(e) => setBody(e.target.value)}/>
            <button>add note</button>
        </form>
        </>
    )
};
export {AddNoteForm as default}