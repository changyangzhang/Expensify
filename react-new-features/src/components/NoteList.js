import Note from "./Note";
import React, { useContext } from "react";
import NotesContext from '../context/notes-contect';

const NoteList = () =>{
    const { notes } = useContext(NotesContext);

    return notes.map((note)=>(
        <Note key={note.title} note={note}/>
    ))
};

export { NoteList as default }