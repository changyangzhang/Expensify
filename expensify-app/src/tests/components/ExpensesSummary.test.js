import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from "../../components/ExpensesSummary";

test('should render ExpensesSummary with 1 expnse', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multi expnses', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={15641656516}/>);
    expect(wrapper).toMatchSnapshot();
});