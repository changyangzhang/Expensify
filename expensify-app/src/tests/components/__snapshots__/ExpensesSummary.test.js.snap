// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expnse 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
         
        2.35 kr
         
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with multi expnses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling 
      <span>
         
        156,416,565.16 kr
         
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
